robottelo/cli/factory.py

```diff
@@ -3635,3 +3715,377 @@ def add_role_permissions(role_id, resource_permissions):
+        'output': temp_virt_who_deploy_file_path
+    })
+    download_file(
+        remote_file=temp_virt_who_deploy_file_path,
+        local_file=temp_virt_who_deploy_file_path,
+        hostname=settings.server.hostname
+    )
+    upload_file(
+        local_file=temp_virt_who_deploy_file_path,
+        remote_file=temp_virt_who_deploy_file_path,
+        hostname=virt_who_vm.ip_addr
+    )
+    # ensure the virt-who config deploy script is executable
+    result = virt_who_vm.run('chmod +x {0}'.format(
+        temp_virt_who_deploy_file_path))
+    if result.return_code != 0:
+        raise CLIFactoryError(
+            u'Virt-Who failed to set deployment script as executable:\n{}'
+            .format(result.stderr)
+        )
+    # execute the deployment script
+    result = virt_who_vm.run('{0}'.format(temp_virt_who_deploy_file_path))
+    if result.return_code != 0:
+        raise CLIFactoryError(
+            u'Virt-Who deployment script failure:\n{}'.format(result.stderr)
+        )
+    # after this step, we should have virt-who service installed and started
+    if exec_one_shot:
+        # some time to be sure that the virt-who generated the report, needed
+        # to force a one shot report, for this we have to stop the virt-who
+        # service
+        result = virt_who_vm.run('service virt-who stop')
+        if result.return_code != 0:
+            raise CLIFactoryError(
+                u'Virt-Who failed to stop the virt-who service:\n{}'
+                .format(result.stderr)
+            )
+        result = virt_who_vm.run('virt-who --one-shot')
+        if result.return_code != 0:
+            raise CLIFactoryError(
+                u'Virt-Who failed when executing virt-who --one-shot:\n{}'
+                .format(result.stderr)
+            )
+        result = virt_who_vm.run('service virt-who start')
+        if result.return_code != 0:
+            raise CLIFactoryError(
+                u'Virt-Who failed to start the virt-who service:\n{}'
+                .format(result.stderr)
+            )
+    # after this step the hypervisor as a content host should be created
+    # do not confuse virt-who host with hypervisor host as they can be
+    # diffrent hosts and as per this setup we have only registered the virt-who
+    # host, the hypervisor host should registered after virt-who send the
+    # first report when started or with one shot command
+    # the virt-who hypervisor will be registered to satellite with host name
+    # like "virt-who-{hypervisor_hostname}-{organization_id}"
+    virt_who_hypervisor_hostname = (
+        'virt-who-{0}-{1}'.format(hypervisor_hostname, org['id']))
+    # find the registered virt-who hypervisor host
+    org_hosts = Host.list({
+        'organization-id': org['id'],
+        'search': 'name={0}'.format(virt_who_hypervisor_hostname)
+    })
+    if len(org_hosts) == 0:
+        raise CLIFactoryError(
+            u'Virt-Who failed to find hypervisor host:\n{}'
```

Comment:
nitpick: drop `virt-who` word, it's misleading, i'd spent extra time looking on failure trying to figure out which virt-who host failed to find/connect hypervisor host, it's satellite who failed to find host